src/main/scala/diplomaticobjectmodel/logicaltree/LogicalTrees.scala

```diff
@@ -32,45 +33,19 @@ class CLINTLogicalTreeNode(device: SimpleDevice, f: => OMRegisterMap) extends Lo
   }
 }
 
-class DebugCSRsLogicalTreeNode(
-  device: SimpleDevice,
-  f: => OMRegisterMap,
-  p: Parameters
-) extends LogicalTreeNode {
-  def getOMDebug(resourceBindings: ResourceBindings): Seq[OMComponent] = {
-    val memRegions: Seq[OMMemoryRegion] = DiplomaticObjectModelAddressing
-      .getOMMemoryRegions("Debug", resourceBindings, Some(f))
-    val cfg: DebugModuleParams = p(DebugModuleParams)
-
-    Seq[OMComponent](
-      OMDebugCSRs(
-        nDebugScratchRegisters = 0,
-        nTriggers = 0,
-        supportedOMTriggerTypes = Nil, // List[List[OMTriggerType]], // Each trigger could support different types
-        nTriggerChainDepth = 0,
-        dscrXdebugver = 0,
-        modeMTdataAccessible = Nil, // List[Boolean],
-        hasMcontrolHit = Nil, //  List[Boolean],
-        supportedMcontrolActions = Nil //  List[TriggerMatchAction], // What to do if trigger hits
-      )
-    )
-  }
-
-  def getOMComponents(resourceBindingsMap: ResourceBindingsMap, components: Seq[OMComponent]): Seq[OMComponent] = {
-    DiplomaticObjectModelAddressing.getOMComponentHelper(device, resourceBindingsMap, getOMDebug)
-  }
-}
-
 class DebugLogicalTreeNode(
   device: SimpleDevice,
-  f: => OMRegisterMap,
-  p: Parameters,
-  m: () => TLDebugModule
-) extends LogicalTreeNode {
+  dmOuter: () => TLDebugModuleOuterAsync,
```

Comment:
why do you want the `Async` versions of these? Why not just pass in dmOuter.dmOuter in the calling function?

Reply:
There is an eval order problem. I should have saved the exception message.